libr/util/ht.c

```diff
@@ -102,9 +98,10 @@ static const struct {
     { 2147483648ul,	2362232233ul,	2362232231ul}
 };
 
-#define entry_is_free(x) (!x || !x->data)
-#define entry_is_deleted(x) x->data==&deleted_data
-#define entry_is_present(x) (x->data && x->data != &deleted_data)
+#define DELETED_HASH ((ut32)~0)
+#define entry_is_free(x) (!x->hash && !x->data)
+#define entry_is_deleted(x) (x->hash == DELETED_HASH && !x->data)
```

Comment:
The old code checked `!x`, but I believe it is unnecessary. The `entry_is_{free,deleted,present}` macros are only visible in `libr/util/ht.c` where they are internally used on the following lines:
```c
116 	hash_address = hash % ht->size;
117 	do {
118 		RHTE *entry = ht->table + hash_address;
119 		if (entry_is_free (entry))
120 			return NULL;
121 		if (entry_is_present (entry) && entry->hash == hash)
122 			return entry;
...
146 	for (e = old_ht.table; e != old_ht.table + old_ht.size; e++) {
147 		if (entry_is_present (e))
148 			ht_(insert) (ht, e->hash, e->data);
...
200 	hash_address = hash % ht->size;
201 	do {
202 		RHTE *entry = ht->table + hash_address;
203 		utH double_hash;
204
205 		if (!entry_is_present (entry)) {
206 			if (entry_is_deleted (entry))
```
Thus, the input parameter `x` can be NULL only if we have a hashtable with a base address of 0x0 (in addition to `hash%ht->size==0`), which should not occur in any circumstances. Intuitively, `x` is a pointer to an entry in a dynamically allocated hashtable, so `x == NULL` is a nonsensical situation.

Is there any benefit in having`entry_is_free(NULL) == true`?